tox.ini

```diff
@@ -247,12 +247,9 @@ deps =
   # needed for tracecontext
   aiohttp~=3.6
   # needed for example trace integration
-  flask~=1.1
+  flask~=3.0
```

Comment:
I believe the flask instrumentation doesn't support [flask 3.X](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/instrumentation/opentelemetry-instrumentation-flask/pyproject.toml#L38). Is this upgrade absolutely necessary?

Reply:
We have to upgrade Flask, but I think it can be 2.x. I upgraded to 3 just because it is the newest version. I'll try to make it work with Flask 2.

Reply:
Done! Passed all tests with Flask 2.